include/allegro5/platform/aintwin.h

```diff
@@ -283,6 +283,11 @@ void _al_win_touch_input_handle_end(int id, size_t timestamp, float x, float y,
 void _al_win_touch_input_handle_move(int id, size_t timestamp, float x, float y, bool primary, ALLEGRO_DISPLAY_WIN *win_disp);
 void _al_win_touch_input_handle_cancel(int id, size_t timestamp, float x, float y, bool primary, ALLEGRO_DISPLAY_WIN *win_disp);
 
+/* Helpers for getting Windows system errors 
+ * May be called from addons 
+ */
+AL_FUNC(const char*, _al_win_error, (DWORD));
```

Comment:
Doesn't look like these are thread safe (on the account of using static buffers). Is that a concern?

Reply:
They weren't thread-safe before, if I understood correctly. However if we're changing the code we might as well make it robust - I don't suppose these functions need to be fast.
I don't know off hand, but I bet you do -  what does the rest of Allegro do where it needs to avoid a static buffer?
* Allocate memory
* User supplies a buffer
* Something else?

Reply:
Ah, okay, if they weren't threadsafe before then we're not making things worse. In fact, this is better now because this logic is centralized, and it is possible to fix it without too much hunting.

I think I'd prefer allocating memory. This is not something you need to do though for this pull request to go in.

Reply:
OK.